### PR TITLE
fix: APP-3098 - resolved QA comments

### PR DESCRIPTION
**Pull request by sepehr2github**
## Description

- Fix Update metadata
- Fix avatar pinning

Task: [APP-3098](https://aragonassociation.atlassian.net/browse/APP-3098)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I have tested my code on the test network.

[APP-3098]: https://aragonassociation.atlassian.net/browse/APP-3098?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/services/ipfs/ipfsService.ts [hunk 1]
```diff
@@ -22,40 +22,59 @@ class IpfsService {
...
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          pinataOptions: {
+            cidVersion: this.CIDVersion,
+          },
+          pinataContent: processedData,
+        }),
+      });
+    }
```
Comment (thekidnamedkd) [inline src/services/ipfs/ipfsService.ts]: Could we do something with a reusable enum type here for the pinData and processData?

```ts
enum DataType {
  File = 'file',
  JSON = 'json',
}
```

File: src/utils/library.ts [hunk 2]
```diff
@@ -395,6 +400,24 @@ export function decodeGaslessSettingsToAction(
...
 
+export async function updateDaoMetadataAction(
+  data: Uint8Array
+): Promise<DaoMetadata | undefined> {
+  await Uint8ArraySchema.strict().validate(data);
+  const daoInterface = DAO__factory.createInterface();
+  const hexBytes = bytesToHex(data);
+  const expectedFunction = daoInterface.getFunction('setMetadata');
+  const result = daoInterface.decodeFunctionData(expectedFunction, hexBytes);
+  const metadataUri = toUtf8String(result[0]);
```
Comment (thekidnamedkd) [inline src/utils/library.ts:411]: can we be sure this is always the Uri at the 0 index?

Comment (sepehr2github) [reply src/utils/library.ts:411]: Yes, this was the function I brought in from SDK

**Review by thekidnamedkd — APPROVED** [review]
Looking good! Left a couple comments.

File: src/utils/contract.ts [hunk 3]
```diff
@@ -2,12 +2,19 @@
...
   a.type === 'function' &&
   (a.stateMutability === 'payable' || a.stateMutability === 'nonpayable') &&
   a.name.toLowerCase().includes(search.toLowerCase());
 
+export const EMPTY_DAO_METADATA_LINK: DaoMetadata = {
+  name: '(the DAO has no metadata)',
+  description: '(the DAO did not define any content)',
+  links: [],
+};
+
```
Comment (thekidnamedkd) [inline src/utils/contract.ts:17]: Where is this getting consumed or used as a fallback?

File: src/utils/proposals.ts [hunk 4]
```diff
@@ -1196,13 +1196,10 @@ export async function getDecodedUpdateActions(
  * @param avatar
  * @param client
  */
-export const pinAvatar = async (avatar: Blob, client: Client) => {
+export const pinAvatar = async (avatar: Blob) => {
   try {
-    const daoLogoBuffer = await readFile(avatar);
```
Comment (thekidnamedkd) [inline src/utils/proposals.ts:1201]: We don't need to read the file anymore before we process?

File: src/utils/contract.ts (hunk 3 above)
Comment (sepehr2github) [reply src/utils/contract.ts:17]: useDaoDetails for fetching Dao metadata

File: src/utils/proposals.ts (hunk 4 above)
Comment (sepehr2github) [reply src/utils/proposals.ts:1201]: Nope, the new pinata end point will take care of that so we can directly write the file. You can test it yourself as well

File: src/services/ipfs/ipfsService.ts (hunk 1 above)
Comment (thekidnamedkd) [reply src/services/ipfs/ipfsService.ts]: Also we know for certain that it's a binary of only file or JSON? Wary of the single else block for everything 'else' before POST call.

**Pull request merged**